# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.71; am-1.16; lt-2.4.7)
libtoolize -c -f; aclocal; autoconf; autoheader; automake -a -c
./configure --disable-static --prefix=

# build
make

# install
install -d $DESTDIR/include/libpng16
install -m 644 png.h $DESTDIR/include/libpng16
ln -sf libpng16/png.h $DESTDIR/include/png.h
install -m 644 pngconf.h $DESTDIR/include/libpng16
ln -sf libpng16/pngconf.h $DESTDIR/include/pngconf.h
install -m 644 pnglibconf.h $DESTDIR/include/libpng16
ln -sf libpng16/pnglibconf.h $DESTDIR/include/pnglibconf.h
install -d $DESTDIR/lib
install .libs/libpng16.so.16.* $DESTDIR/lib/libpng16.so.16
ln -sf libpng16.so.16 $DESTDIR/lib/libpng16.so
ln -sf libpng16.so $DESTDIR/lib/libpng.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 libpng16.pc $DESTDIR/lib/pkgconfig
ln -sf libpng16.pc $DESTDIR/lib/pkgconfig/libpng.pc

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/libpng-build.pc << EOF
Name: libpng
Version: 1.6.40
Description: Portable Network Graphics (PNG) library
Requires: $REQUIRES

devel=\\
/include/png.h \\
/include/pngconf.h \\
/include/pnglibconf.h \\
/include/libpng16/png.h \\
/include/libpng16/pngconf.h \\
/include/libpng16/pnglibconf.h \\
/lib/libpng.so \\
/lib/libpng16.so \\
/lib/pkgconfig/libpng.pc \\
/lib/pkgconfig/libpng16.pc

exec=\\
/lib/libpng16.so.16
EOF

exit
--- libpng-1.6.40.orig/pngget.c
+++ libpng-1.6.40/pngget.c
@@ -1026,7 +1026,8 @@
    png_debug1(1, "in %s retrieval function", "PLTE");
 
    if (png_ptr != NULL && info_ptr != NULL &&
-       (info_ptr->valid & PNG_INFO_PLTE) != 0 && palette != NULL)
+       (info_ptr->valid & PNG_INFO_PLTE) != 0 &&
+       palette != NULL && num_palette != NULL)
    {
       *palette = info_ptr->palette;
       *num_palette = info_ptr->num_palette;
